src/wifi.cpp

```diff
@@ -0,0 +1,694 @@
+// Resources:
+// - Create a basic POSIX TPC/IP server in C:
+//   https://medium.com/@coderx_15963/basic-tcp-ip-networking-in-c-using-posix-9a074d65bb35
+// - How to connect your ESP32 to Wifi and an existing POSIX TCP/IP server:
+//   https://www.youtube.com/watch?v=_dRrarmQiAM
+//   https://github.com/espressif/esp-idf/blob/v5.3.1/examples/wifi/getting_started/station/main/station_example_main.c
+//   https://docs.espressif.com/projects/esp-idf/en/stable/esp32/api-reference/network/esp_netif.html
+//   https://docs.espressif.com/projects/esp-idf/en/stable/esp32/api-reference/system/esp_event.html
+//   https://docs.espressif.com/projects/esp-idf/en/stable/esp32/api-reference/network/esp_wifi.html
+
+#include "wifi.h"
+
+#if WIFI_ENABLED
+
+// Include FreeRTOS common header
+#include "freertos/FreeRTOS.h"
+// Include FreeRTOS task API
+#include "freertos/task.h"
+// Include FreeRTOS event group API
+#include "freertos/event_groups.h"
+// Import light-weight IP socket API
+#include "lwip/sockets.h"
+// Include ESP WiFi API
+#include "esp_wifi.h"
+// Include ESP event API
+#include "esp_event.h"
+
+// Include custom Menu class implementation
+#include "menu.h"
+
+// ====================================== //
+// Define useful constants and data types //
+// ====================================== //
+
+// Define bitmask for success in WiFi event loop
+#define WIFI_CONNECTED_BIT BIT0
+// Define bitmask for failure in WiFi event loop
+#define WIFI_FAIL_BIT BIT1
+// Define the max number of retries allowed to try to connect to one AP in a row
+#define NUM_MAX_WIFI_CONNECT_RETRIES 5
+
+// Define the number of currently supported TCP commands
+#define NUM_TCP_COMMANDS 3
+
+// Define, when receiving a TCP packet, what special strings should cause what actions
+typedef struct tcp_command_s {
+    // The string, that if the TCP packet matches, should trigger an action
+    const char* command;
```

Comment:
To be consistent with the rest of the code:
`const char *command;`

Reply:
addressed in: https://github.com/jCallon/squirt/pull/1/commits/edb88a636d2cf54a42f970d190cd054c04993412